otherlibs/stdune/test/pp_tests.ml

```diff
@@ -0,0 +1,188 @@
+open Stdune
+
+let test_truncate length pp =
+  let pp = Pp.truncate length pp in
+  Format.printf "%s" (Pp.to_dyn Dyn.opaque pp |> Dyn.to_string)
+;;
+
+let test_truncate_and_print length pp =
+  let pp = Pp.truncate length pp in
+  Format.printf "%a" Pp.to_fmt pp
```

Comment:
Can you write an assert that makes sure we don't print:

1. More than one line
2. We don't go over the length hard limit